packages/frontend-2/components/server-management/DeleteInvitationDialog.vue

```diff
@@ -0,0 +1,151 @@
+        <strong>cannot</strong>
+        be undone.
+      </p>
+    </div>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import { useMutation } from '@vue/apollo-composable'
+import { LayoutDialog } from '@speckle/ui-components'
+import { InviteItem } from '~~/lib/server-management/helpers/types'
+import { adminDeleteInvite } from '~~/lib/server-management/graphql/mutations'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import {
+  ROOT_QUERY,
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage,
+  modifyObjectFields
+} from '~~/lib/common/helpers/graphql'
+import { AdminInviteList } from '~~/lib/common/generated/gql/graphql'
+
+const props = defineProps<{
+  open: boolean
+  title: string
+  invite: InviteItem | null
+}>()
+
+const emit = defineEmits<{
+  (e: 'update:open', val: boolean): void
+}>()
+
+const { triggerNotification } = useGlobalToast()
+const { mutate: adminDeleteMutation } = useMutation(adminDeleteInvite)
+
+const isOpen = computed({
+  get: () => props.open,
+  set: (newVal) => emit('update:open', newVal)
+})
+
+const deleteConfirmed = async () => {
+  const inviteId = props.invite?.id
+  if (!inviteId) {
+    return
+  }
+
+  const result = await adminDeleteMutation(
+    {
+      inviteId
+    },
+    {
+      update: (cache, { data }) => {
+        if (data?.inviteDelete) {
+          // Remove invite from cache
+          const cacheId = getCacheId('ServerInvite', inviteId)
+          cache.evict({
+            id: cacheId
+          })
+
+          // Modify 'admin' field of ROOT_QUERY so that we can modify all `inviteList` instances
+          modifyObjectFields<undefined, { [key: string]: AdminInviteList }>(
+            cache,
+            ROOT_QUERY,
+            (_fieldName, _variables, value, details) => {
+              // Find all `inviteList` fields (there can be multiple due to differing variables)
+              const inviteListFields = Object.keys(value).filter(
+                (k) =>
+                  details.revolveFieldNameAndVariables(k).fieldName === 'inviteList'
+              )
+
+              // Being careful not to mutate original `value`
+              const newVal: typeof value = { ...value }
+
+              // Iterate over each and adjust `items` and `totalCount`
+              for (const field of inviteListFields) {
+                const oldItems = value[field]?.items || []
+                const newItems = oldItems.filter((i) => i.__ref !== cacheId)
+                const removedCount = oldItems.length - newItems.length
+
+                newVal[field] = {
+                  ...value[field],
+                  items: newItems,
+                  totalCount: Math.max(
+                    0,
+                    (value[field]?.totalCount || 0) - removedCount
+                  )
+                }
+              }
+
+              return newVal
+            },
+            { fieldNameWhitelist: ['admin'] }
```

Comment:
This bit is the one that ensures we only update Query.admin, not some other ROOT_QUERY fields.